### PR TITLE
Redis sdb mod

### PR DESCRIPTION
**Pull request by mirceaulinic**
### What does this PR do?

Adds an SDB module for Redis.

### Tests written?

No - but happy to provide: I'm unsure how to mock a connection to a Redis server, without having the Redis server.

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/sdb/redis_sdb.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+def set_(key, value, profile=None):
+    '''
+    Set a value into the Redis SDB.
+    '''
+    if not profile:
+        return False
+    redis_kwargs = profile.copy()
+    redis_kwargs.pop('driver')
+    redis_conn = redis.Redis(**redis_kwargs)
```
Comment (gtmanfred) [inline salt/sdb/redis_sdb.py]: Fyi, we pack the minion modules into sdb.

https://github.com/saltstack/salt/blob/develop/salt/loader.py#L892

So, you should just be able to do `__salt__['redis.set'](key, value, **redis_kwargs)`

**Review by gtmanfred — CHANGES_REQUESTED** [review]
Just a small change to use the already written redis module, instead of importing redis here.

Thanks,
Daniel

File: salt/sdb/redis_sdb.py (hunk 1 above)
Comment (mirceaulinic) [reply salt/sdb/redis_sdb.py]: I tried on 2017.7 and it doesn't work like that. Did that change in 2018.3?

Either way, the Redis module only accepts `host`, `password`, `port`, and `db` as arguments, while there are many more available (and with the code I proposed we don't impose any restriction - some of them are very important!) - see https://redis-py.readthedocs.io/en/latest/#contents. What do you think?

Comment (gtmanfred) [reply salt/sdb/redis_sdb.py]: OK, those should also be added to the redis module, but we don't have to do that now, but this should still use the StrictRedis class

Comment (mirceaulinic) [reply salt/sdb/redis_sdb.py]: Yes, good catch, should use `StrictRedis` indeed.

Comment (mirceaulinic) [reply salt/sdb/redis_sdb.py]: Pushed https://github.com/saltstack/salt/pull/48239/commits/3f17add324361e177fc9eab8bcfb98cf91109e18 to solve this.

**Review by gtmanfred — APPROVED** [review]

File: salt/sdb/redis_sdb.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      host: 127.0.0.1
+      port: 6379
+      password: pass
+      db: 1
+
+The ``driver`` refers to the Redis module, all other options are optional.
+For option details see: https://redis-py.readthedocs.io/en/latest/.
+
+'''
+from __future__ import absolute_import
```
Comment (rallytime) [inline salt/sdb/redis_sdb.py]: We also need to import `print_function` and `unicode_literals` here for PY3 compatibility.

**Review by rallytime — CHANGES_REQUESTED** [review]
One small request from me. :)

This could use a `.. versionadded:: Fluorine` tag as well.

File: salt/sdb/redis_sdb.py (hunk 2 above)
Comment (mirceaulinic) [reply salt/sdb/redis_sdb.py]: Good points - added: https://github.com/saltstack/salt/pull/48239/commits/30cd544d7ebc2491db6034c4215d59b73db58def

Regarding `print_function` and `unicode_literals`: I've seen this has been added across all the modules, and I'm afraid I didn't understand that - could you please explain why is `print_function` (for example) needed although the module doesn't call the `print()` function? Thanks!

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
Thanks @mirceaulinic!

**Pull request merged**